flang/module/cudadevice.f90

```diff
@@ -92,5 +92,31 @@ attributes(device) subroutine threadfence_system()
     end function
   end interface
   public :: __fadd_ru
+
+  ! Atomic Operations
   
+  interface atomicadd
+    attributes(device) pure integer function atomicaddi(address, val)
+  !dir$ ignore_tkr (rd) address, (d) val
```

Comment:
I see that one is allowed to call `atomicadd` with an address of any rank ("r"). What is the semantic of calling them with an array? Is it that the first element is modified, or that all elements are modified?

Reply:
Brent mentioned 1 element arrays but could not be definitive. I can remove it and maybe we will learn something.